### PR TITLE
Wip/hughsie/logi tap misc

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@vcdmp can you just check I've not broken anything please.

**Review by superm1 — APPROVED** [review]

**Comment by vcdmp** [comment]
> @vcdmp can you just check I've not broken anything please.

Running into following error, never saw before.
root@vc:/home/vc/repo/fwupd/build# FU_UDEV_DEVICE_DEBUG=1 FWUPD_VERBOSE=1 FWUPD_DEVICE_LIST_VERBOSE=1 FU_USB_DEVICE_DEBUG=1 G_MESSAGES_DEBUG=all ./src/fwupdtool --verbose --verbose --cleanup --plugin-whitelist logitech_tap get-devices
(process:120365): GLib-GIO-DEBUG: 09:43:29.722: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
(process:120365): dconf-DEBUG: 09:43:29.722: watch_fast: "/system/proxy/" (establishing: 0, active: 0)
(process:120365): dconf-DEBUG: 09:43:29.722: watch_fast: "/system/proxy/http/" (establishing: 0, active: 0)
(process:120365): dconf-DEBUG: 09:43:29.722: watch_fast: "/system/proxy/https/" (establishing: 0, active: 0)
(process:120365): dconf-DEBUG: 09:43:29.722: watch_fast: "/system/proxy/ftp/" (establishing: 0, active: 0)
(process:120365): dconf-DEBUG: 09:43:29.722: watch_fast: "/system/proxy/socks/" (establishing: 0, active: 0)
(process:120365): dconf-DEBUG: 09:43:29.722: unwatch_fast: "/system/proxy/" (active: 0, establishing: 1)
(process:120365): dconf-DEBUG: 09:43:29.722: unwatch_fast: "/system/proxy/http/" (active: 0, establishing: 1)
(process:120365): dconf-DEBUG: 09:43:29.722: unwatch_fast: "/system/proxy/https/" (active: 0, establishing: 1)
(process:120365): dconf-DEBUG: 09:43:29.722: unwatch_fast: "/system/proxy/ftp/" (active: 0, establishing: 1)
(process:120365): dconf-DEBUG: 09:43:29.722: unwatch_fast: "/system/proxy/socks/" (active: 0, establishing: 1)
(process:120365): GLib-DEBUG: 09:43:29.722: posix_spawn avoided (fd close requested) 
(process:120365): GLib-GIO-DEBUG: 09:43:29.724: _g_io_module_get_default: Found default implementation libproxy (GLibproxyResolver) for ‘gio-proxy-resolver’
(fwupdtool:120365): GLib-DEBUG: 09:43:29.725: setenv()/putenv() are not thread-safe and should not be used after threads are created
16:43:29.725 FuDebug              verbose to debug (on console 1)
16:43:29.725 FuEngine             starting fwupd 1.9.1…
16:43:29.725 FuMain               locked /run/lock/fwupdtool
(fwupdtool:120365): dconf-DEBUG: 09:43:29.728: watch_established: "/system/proxy/" (establishing: 0)
(fwupdtool:120365): dconf-DEBUG: 09:43:29.729: watch_established: "/system/proxy/http/" (establishing: 0)
(fwupdtool:120365): dconf-DEBUG: 09:43:29.729: watch_established: "/system/proxy/https/" (establishing: 0)
(fwupdtool:120365): dconf-DEBUG: 09:43:29.729: watch_established: "/system/proxy/ftp/" (establishing: 0)
(fwupdtool:120365): dconf-DEBUG: 09:43:29.729: watch_established: "/system/proxy/socks/" (establishing: 0)
16:43:29.731 FuMain               failed to stop daemon: failed to find fwupd.service: GDBus.Error:org.freedesktop.systemd1.NoSuchUnit: Unit fwupd.service not loaded.
Loading…                 [ -                                     ]16:43:29.731 FuConfig             trying to load config values from /home/vc/.root/etc/fwupd/fwupd.conf
16:43:29.731 FuCommon             reading /home/vc/.root/etc/fwupd/fwupd.conf with 119 bytes
16:43:29.731 FuCommon             reading /home/vc/.root/etc/fwupd/daemon.conf with 2563 bytes
16:43:29.731 FuCommon             reading /home/vc/.root/etc/fwupd/msr.conf with 94 bytes
16:43:29.731 FuCommon             reading /home/vc/.root/etc/fwupd/redfish.conf with 486 bytes
16:43:29.731 FuCommon             reading /home/vc/.root/etc/fwupd/thunderbolt.conf with 349 bytes
16:43:29.731 FuCommon             reading /home/vc/.root/etc/fwupd/uefi_capsule.conf with 818 bytes
16:43:29.731 FuConfig             trying to load config values from /home/vc/.root/var/etc/fwupd/fwupd.conf
16:43:29.731 FuConfig             not migrating default value of [fwupd] ApprovedFirmware=
16:43:29.731 FuConfig             not migrating default value of [fwupd] ArchiveSizeMax=0
16:43:29.731 FuConfig             not migrating default value of [fwupd] BlockedFirmware=
16:43:29.731 FuConfig             not migrating default value of [fwupd] DisabledDevices=
16:43:29.731 FuConfig             not migrating default value of [fwupd] EnumerateAllDevices=false
16:43:29.731 FuConfig             not migrating default value of [fwupd] EspLocation=
16:43:29.731 FuConfig             not migrating default value of [fwupd] HostBkc=
16:43:29.731 FuConfig             not migrating default value of [fwupd] IdleTimeout=7200
16:43:29.731 FuConfig             not migrating default value of [fwupd] ShowDevicePrivate=true
16:43:29.731 FuConfig             not migrating default value of [fwupd] TrustedUids=
16:43:29.731 FuConfig             not migrating default value of [fwupd] UpdateMotd=true
16:43:29.731 FuConfig             not migrating default value of [fwupd] UriSchemes=
16:43:29.731 FuConfig             not migrating default value of [fwupd] VerboseDomains=
16:43:29.731 FuConfig             not migrating default value of [redfish] IpmiDisableCreateUser=False
16:43:29.731 FuConfig             not migrating default value of [redfish] ManagerResetTimeout=1800

Failed to load config: did not write /home/vc/.root/etc/fwupd/fwupd.conf file

File: plugins/logitech-tap/fu-logitech-tap-hdmi-device.c [hunk 1]
```diff
@@ -376,72 +370,56 @@ fu_logitech_tap_hdmi_device_write_firmware(FuDevice *device,
...
 
-	/* map endian data. MajorVersion byte2 3&2, MinorVersion bytes 5&4, BuildVersion bytes 7&6
-	 */
-	if (!fu_memread_uint16_safe(query_data,
-				    sizeof(query_data),
-				    0x2,
-				    &major,
-				    G_BIG_ENDIAN,
-				    error))
+	/* MajorVersion byte2 3&2, MinorVersion bytes 5&4, BuildVersion bytes 7&6 */
```
Comment (vcdmp) [inline plugins/logitech-tap/fu-logitech-tap-hdmi-device.c:415]: Sorry my typo 'bytes' not 'byte2'

File: plugins/logitech-tap/fu-logitech-tap-sensor-device.c [hunk 2]
```diff
@@ -370,22 +349,12 @@ fu_logitech_tap_sensor_device_init(FuLogitechTapSensorDevice *self)
...
-{
-	G_OBJECT_CLASS(fu_logitech_tap_sensor_device_parent_class)->finalize(object);
 }
 
 static void
 fu_logitech_tap_sensor_device_class_init(FuLogitechTapSensorDeviceClass *klass)
 {
-	GObjectClass *object_class = G_OBJECT_CLASS(klass);
 	FuDeviceClass *klass_device = FU_DEVICE_CLASS(klass);
-	object_class->finalize = fu_logitech_tap_sensor_device_finalize;
```
Comment (vcdmp) [inline plugins/logitech-tap/fu-logitech-tap-sensor-device.c:388]: If finalize not required, remove it from fu-logitech-tap-hdmi-device.c as well!

**Review by vcdmp — APPROVED** [review]
Few minor observations.

File: plugins/logitech-tap/fu-logitech-tap-hdmi-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/logitech-tap/fu-logitech-tap-hdmi-device.c:415]: Fixed, thanks.

File: plugins/logitech-tap/fu-logitech-tap-sensor-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/logitech-tap/fu-logitech-tap-sensor-device.c:388]: Ack, done.

**Pull request merged**